### PR TITLE
Php2branch >tlr1revbranch

### PR DESCRIPTION
**Pull request by php-2**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+include_once '../../../../master.inc.php';
+include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/app/proposal/clsproposal.class.php';
+require_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
+    $response = array();
+    if($_SERVER['REQUEST_METHOD']=='POST')
+    {
+        if (__DEBUG__) {echo '<br>if post request method</br>';}
+        
+        if(isset($_POST['id']))
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php]: don't use _POST parameters
park it at local parameters and use local parameters instead.

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        {
+            $obj= new clsproposal();
+            if (__DEBUG__) {echo '<br>check object from clsproposal?? </br>';}
+                if (__DEBUG__) {echo '<br>apigetproposaldata.php has been called</br>';}
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php]: why 2 debug statement subsequently?
use either one.

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                if (__DEBUG__) {echo '<br/>Result set passed to apigetproposaldata.php<br/>';}
+                $response['error']= false;
+                $response['empty']=false;
+                $rs=$obj->fnGetProposalData($_POST['id']);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apigetproposaldata.php]: don't use POST parameter directly.

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+$response = array();
+    if($_SERVER['REQUEST_METHOD']=='POST')
+    {
+        if (__DEBUG__) {echo '<br>if post request method</br>';}
+        
+        if(isset($_POST['id']))
+        {
+            $obj = new clsproposal();
+            if (__DEBUG__) {echo '<br>check object from clsproposal?? </br>';}
+                if (__DEBUG__) {echo '<br>apivalidateproposal.php has been called</br>';}
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: why 2 debug lines?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                $response['empty']="false";
+                $response['action']=$obj->fnValidateProposal($_POST['id'],$_POST['userid']);
+       }
+        else{
+                $response["error"]=NO_PAR_ERROR; //"no parammeters passed";	
+            }
+    }
+        else 
+        {
+            $response["error"]=NO_POSTMET_ERROR;  //"method is not post";
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: NO_POSTMET_ERROR
what meaning of MET?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+if (__DEBUG__) { echo '</br>clsProposal has been included</br>'; }
+class clsproposal //extends DolibarrApi
+{ 
+        public $db = null;
+
+	function __construct() {
+            global  $db,$conf;
+            // Create connection
+               $this->db=$db;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php]: why u use local parameter and not the dolibarr global parametr after you define.

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	}
+        public function fnGetProposalData($id)
+        {
+            try
+            { 
+                if (__DEBUG__) {echo '</br>fnAppUser_GetProposalData() begin ? </br>';}
+                  $sql = "SELECT p.rowid  ,p.ref , s.nom as ref_client ,p.datec as date_creation ,p.date_valid as date_validate ,p.fin_validite as finish_date ,p.fk_statut as status ,ROUND(p.price, 2) AS price  ,p.model_pdf ,p.fk_currency as currency ,ROUND(p.total , 2) as total,prod.rowid as product_id ,prod.ref as description ,prod.label as description_label  ,ROUND(prod.cost_price , 2) as cost_price ,pr.qty as Qty_Reduc ,ROUND(pr.total_tva, 2) as totoal_net ,ROUND(pr.tva_tx, 2) as Sales_tax ,ROUND(pr.buy_price_ht, 2) as totoal_net ,ROUND(pr.total_ttc, 2) as totoal_netinCruency " ;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php]: round currency is not necessary to be 2 
it should follow the currency decimal point setup.

I believe this info is loaded at dolibarr global parameters

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                if (__DEBUG__) {echo '</br>fnUndoValidateProposal() begin ? </br>';}
+                  $date_now=date("Y-m-d H:i:s");
+                  $sql = "UPDATE ";
+                  $sql .=MAIN_DB_PREFIX."propal ";
+                  $sql .="SET date_valid = NULL, fk_statut = 0, fk_user_valid=NULL ";
+                  $sql .= "WHERE rowid=$id";
+                  $result=$this->db->query($sql);
+                if ($this->db->affected_rows($result) ==1)
+                        {
+                            $obj=2;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php]: why obj here = 2?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?php
+//by :kamal asran php-2 10/1/2017
+include_once '../../../../master.inc.php';
+include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/class/clsorders.class.php';
+require_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php]: I believe you should include the constant first

Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php]: why u use require_once and not include_once?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    $response = array();
+    if($_SERVER['REQUEST_METHOD']=='POST')
+    {
+        if (__DEBUG__) {echo '<br>if post request method</br>';}
+        
+         $objAppUser = new clsorder($db);
+         if (__DEBUG__) {echo '<br>check object from clsorders?? </br>';}
+         $response["error"]=false;
+         $response["empty"]=false;
+         $response["data"]=$objAppUser->fnAppUser_GetAll();
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php]: what AppUser??????????????????!!!!!!!!!!

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+//by :kamal asran php-2 10/1/2017
+include_once '../../../../master.inc.php';
+include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/class/clsorders.class.php';
+require_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
+    $response = array();
+    if($_SERVER['REQUEST_METHOD']=='POST')
+    {
+        if (__DEBUG__) {echo '<br>if post request method</br>';}
+        
+         $objAppUser = new clsorder($db);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php]: clsorder
should be clsOrder

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<?php
+include_once '../../../../master.inc.php';
+include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/app/proposal/clsproposal.class.php';
+require_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
+$response = array();
+    if($_SERVER['REQUEST_METHOD']=='POST')
+    {
+        if (__DEBUG__) {echo '<br>if post request method</br>';}
+        
+        if(isset($_POST['id']))
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: don't use POST parameters directly

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            $obj = new clsproposal();
+            if (__DEBUG__) {echo '<br>check object from clsproposal?? </br>';}
+                if (__DEBUG__) {echo '<br>apivalidateproposal.php has been called</br>';}
+                if (__DEBUG__) {echo '<br/>Result set passed to apivalidateproposal.php<br/>';}
+                $response['error']= "false";
+                $response['empty']="false";
+                $response['action']=$obj->fnValidateProposal($_POST['id'],$_POST['userid']);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: don't use POST parameters directly

Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: what userid here means?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+	function __construct() {
+            global  $db,$conf;
+            // Create connection
+               $this->db=$db;
+	}
+        public function fnGetProposalData($id)
+        {
+            try
+            { 
+                if (__DEBUG__) {echo '</br>fnAppUser_GetProposalData() begin ? </br>';}
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php]: fnAppUser_GetProposalData?????

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php (hunk 5 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: MET stands for Method

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php (hunk 13 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/apivlaidateproposal.php]: the id of the logged in user that try to validate the proposal

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php (hunk 8 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/proposal/clsproposal.class.php]: we tried  to return numbers in json to be 1 if validate process is  done and 2 if undo_validate process  was done ,but now i understand you that we should use global variables and define statements in constant file

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php (hunk 9 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/apiGetall.php]: why ? i don,t know the difference between the two  ways .

**Pull request merged**